sample/src/main/java/com/airwallex/paymentacceptance/viewmodel/APIIntegrationViewModel.kt

```diff
@@ -0,0 +1,279 @@
+package com.airwallex.paymentacceptance.viewmodel
+
+import android.app.Activity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.airwallex.android.core.Airwallex
+import com.airwallex.android.core.AirwallexCheckoutMode
+import com.airwallex.android.core.AirwallexPaymentSession
+import com.airwallex.android.core.AirwallexPaymentStatus
+import com.airwallex.android.core.AirwallexRecurringSession
+import com.airwallex.android.core.AirwallexRecurringWithIntentSession
+import com.airwallex.android.core.AirwallexSession
+import com.airwallex.android.core.BillingAddressParameters
+import com.airwallex.android.core.GooglePayOptions
+import com.airwallex.android.core.model.AvailablePaymentMethodType
+import com.airwallex.android.core.model.PaymentConsent
+import com.airwallex.android.core.model.PaymentIntent
+import com.airwallex.android.core.model.PaymentMethod
+import com.airwallex.android.core.model.RetrieveAvailablePaymentConsentsParams
+import com.airwallex.android.core.model.RetrieveAvailablePaymentMethodParams
+import com.airwallex.paymentacceptance.Settings
+import com.airwallex.paymentacceptance.autoCapture
+import com.airwallex.paymentacceptance.nextTriggerBy
+import com.airwallex.paymentacceptance.shipping
+import com.airwallex.paymentacceptance.viewmodel.base.BaseViewModel
+import java.math.BigDecimal
+
+class APIIntegrationViewModel : BaseViewModel() {
+
+    //our SDK offers three modes of payment flow.
+    private var checkoutMode = AirwallexCheckoutMode.PAYMENT
+
+    //create your own Airwallex instance to call the APIs.
+    private var airwallex: Airwallex? = null
+
+    //AirwallexPaymentStatus is the result returned by the payment flow. You can add your own handling logic based on the final result.
+    private val _airwallexPaymentStatus = MutableLiveData<AirwallexPaymentStatus>()
+    val airwallexPaymentStatus: LiveData<AirwallexPaymentStatus> = _airwallexPaymentStatus
+
+    private val _paymentMethodList = MutableLiveData<List<AvailablePaymentMethodType>>()
+    val paymentMethodList: LiveData<List<AvailablePaymentMethodType>> = _paymentMethodList
+
+    private val _paymentConsentList = MutableLiveData<List<PaymentConsent>>()
+    val paymentConsentList: LiveData<List<PaymentConsent>> = _paymentConsentList
+
+    override fun init(activity: Activity) {
+        airwallex = Airwallex(activity)
```

Comment:
Will this potentially cause circular reference? i.e. activity -> viewModel, viewModel -> airwallex -> activity